### PR TITLE
Fix invalidate action button for input dialog

### PR DESCRIPTION
**Pull request by ShurikuS57**
Fix bug for input dialog. When set flag **maxLength** no disabled action button.

### CODE REVIEW

**Comment by afollestad** [comment]
I'm not sure what the actual problem being fixed here is. A better description or screenshots/videos would make this make more sense.

File: input/src/main/java/com/afollestad/materialdialogs/input/InputUtilExt.kt [hunk 1]
```diff
@@ -22,12 +22,15 @@ import com.afollestad.materialdialogs.MaterialDialog
...
 import com.afollestad.materialdialogs.actions.setActionButtonEnabled
 
-internal fun MaterialDialog.invalidateInputMaxLength() {
+internal fun MaterialDialog.invalidateInput(allowEmpty: Boolean) {
   val maxLength = getInputLayout().counterMaxLength
   val currentLength = getInputField().text?.length ?: 0
-  if (maxLength > 0) {
-    setActionButtonEnabled(POSITIVE, currentLength <= maxLength)
-  }
+  val isActionButtonEnable = if (maxLength > 0 && currentLength <= maxLength && !allowEmpty && currentLength > 0) {
```
Comment (afollestad) [inline input/src/main/java/com/afollestad/materialdialogs/input/InputUtilExt.kt]: This should be a when statement, the conditions are rather long

**Comment by ShurikuS57** [comment]
Problem description.

**Comment by afollestad** [comment]
The action button is not disabled with max length, at all? Or at a specific time?

**Comment by ShurikuS57** [comment]
Problem description. 
1. Start origin sample app.
2. Click button "Input + counter"
3. First problem. In code value **allowEmpty=false**, but the **agree** button is available. It is wrong.
4. Second problem. Enter text up to 8 characters. The **agree**  button remains available.
image

**Comment by afollestad** [comment]
Got it, thanks for the additional information.

I want to make sure this doesn't break anything else before I merge

**Pull request closed without merging**